### ISSUE TITLE
Allow using a composite build as a source dependency

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle-native#749]
https://github.com/gradle/gradle/pull/5890 adds support for nested included builds. However, this does not yet work for source dependencies. We should add support for using a composite build as a source dependency.

**Comment by clayreimann** [comment]
I'm confused on what this means:
> add support for using a composite build as a source dependency

How can you create a composite build using composite builds that are not `source dependencies`? If your PR doesn't allow for composite of composite projects, then what exactly does it do? Does it allow for nesting of project [like this answer describes](https://stackoverflow.com/a/46450198/424210)?

**Comment by lacasseio** [comment]
If I'm not mistaken this is adding support for nested composite builds as part of a new feature not yet announced called source dependencies. You can see an [example of how this works in the native samples](https://github.com/gradle/native-samples/tree/master/swift/source-dependencies).

**Comment by lacasseio** [comment]
Correct, the dependency substitution should work correctly with nested included build outside of VCS. If it doesn't work, open an issue so we can look into it.
